xds/src/main/java/io/grpc/xds/ClusterImplLoadBalancer.java

```diff
@@ -386,4 +392,24 @@ public void streamClosed(Status status) {
       };
     }
   }
+
+  private static final class OrcaPerRpcListener implements OrcaPerRequestReportListener {
+
+    private final ClusterLocalityStats stats;
+
+    private OrcaPerRpcListener(ClusterLocalityStats stats) {
+      this.stats = checkNotNull(stats, "stats");
+    }
+
+    /**
+     * Copies {@link MetricReport#getNamedMetrics()} to {@link ClusterLocalityStats} such that it is
+     * included in the snapshot for the LRS report sent to the LRS server.
+     */
+    @Override
+    public void onLoadReport(MetricReport report) {
+      for (Map.Entry<String, Double> entry : report.getNamedMetrics().entrySet()) {
+        stats.recordBackendLoadMetricStats(entry.getKey(), entry.getValue());
```

Comment:
I think we should pass the map `Map<String, Double>` to `recordBackendLoadMetricStats()` to let it optimize the thread synchronization as appropriate. Since `recordBackendLoadMetricStats()` is an internal method, it doesn't hurt anything to pass the Map from what I can tell.

Reply:
Done.
The named metrics map (`Map<String, Double>`) is passed to `recordBackendLoadMetricStats()`.